sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastNestedLoopJoinExec.scala

```diff
@@ -353,7 +353,7 @@ case class BroadcastNestedLoopJoinExec(
     val broadcastedRelation = broadcast.executeBroadcast[Array[InternalRow]]()
```

Comment:
per offline discussion, depending on the semantics we decide, we should revisit whether we'd like to keep the physical plan check here L344